src/python/foglamp/core/browser.py

```diff
@@ -0,0 +1,227 @@
+
+    # Process additional where clause conditions
+    query += _where_clause(request)
+
+    # Add the order by and limit clause
+    limit = __DEFAULT_LIMIT
+    if 'limit' in request.query:
+      limit = request.query['limit']
+
+    orderby = ' order by user_ts limit {0}'.format(limit)
+    query += orderby
+
+    # Select the assets from the readings table
+    rows = await conn.fetch(query);
+    columns = ('timestamp', reading)
+    results = []
+    for row in rows:
+      results.append(dict(zip(columns, row)))
+
+    # Close the connection.
+    await conn.close()
+
+    return web.json_response(results);
+
+async def asset_summary(request):
+    """
+    Browse all the asserts for which we have recorded readings and
+    return a summary for a particulat sensor. The values that are
+    returned are the min, max and average values of the sensor.
+
+    The readings summarised can also be time limited by use of the query
+    parameter seconds=sss. This defines a number of seconds that the reading
+    must have been processed in. Older readings than this will not be summarised.
+
+    The readings summarised can also be time limited by use of the query
+    parameter minutes=mmm. This defines a number of minutes that the reading
+    must have been processed in. Older readings than this will not be summarised.
+
+    The readings summarised can also be time limited by use of the query
+    parameter hours=hh. This defines a number of hours that the reading
+    must have been processed in. Older readings than this will not be summarised.
+
+    Only one of hour, minutes or seconds should be supplied
+
+    Return the result of the Postgres query 
+    select min(reading->>'reading'), max(reading->>'reading'), avg((reading->>'reading')::float) from readings where asset_code = 'asset_code'
+    """
+
+    conn = await asyncpg.connect(__DB_URL)
+    asset_code =  request.match_info.get('asset_code', '')
+    reading =  request.match_info.get('reading', '')
+
+    query = 'select min(reading->>\'{1}\'), max(reading->>\'{1}\'), avg((reading->>\'{1}\')::float) from readings where asset_code = \'{0}\''.format(asset_code, reading)
+
+    query += _where_clause(request)
+    # Select the assets from the readings table
+    row = await conn.fetchrow(query);
+    columns = ('min', 'max', 'average')
+    results = dict(zip(columns, row))
+
+    # Close the connection.
+    await conn.close()
+
+    return web.json_response({ reading : results });
+
+def _where_clause(request):
+    where_clause = ''
+
+    if 'seconds' in request.query:
+      where_clause += ' and user_ts > NOW() - INTERVAL \'{0} seconds\''.format(request.query['seconds'])
+    if 'minutes' in request.query:
+      where_clause += ' and user_ts > NOW() - INTERVAL \'{0} minutes\''.format(request.query['minutes'])
+    if 'hours' in request.query:
```

Comment:
should not these if 'minutes' and if 'hours' be **elif** ?

> +  Note seconds, minutes and hours can not be combined in a URL. If they are then only seconds
> +  will have an effect.

or avoid where_clause +=, just do at return and reverse the checking order..